### PR TITLE
Add tap and unit to ZValidation

### PR DESCRIPTION
**Pull request by nox213**
I think those two are useful for ZValidation.

### CODE REVIEW

File: core/shared/src/main/scala/zio/prelude/ZValidation.scala [hunk 1]
```diff
@@ -334,6 +334,12 @@ sealed trait ZValidation[+W, +E, +A] { self =>
       case (Success(w, _), Failure(w1, e1)) => Failure(w ++ w1, e1)
       case (Success(w, a), Success(w1, b))  => Success(w ++ w1, f(a, b))
     }
+
+  final def tap[W1 >: W, E1 >: E](f: A => ZValidation[W1, E1, Any]): ZValidation[W1, E1, A] =
```
Comment (ghostdogpr) [inline core/shared/src/main/scala/zio/prelude/ZValidation.scala]: I think methods are sorted alphabetically

Comment (sideeffffect) [reply core/shared/src/main/scala/zio/prelude/ZValidation.scala]: Indeed, @nox213 would you mind putting them in the appropriate order? :pray:

Comment (nox213) [reply core/shared/src/main/scala/zio/prelude/ZValidation.scala]: [6ea7901](https://github.com/zio/zio-prelude/pull/1299/commits/6ea79015463d5f2f6d29af1fee4c2dae5a244fdc)
Thanks for the review

**Review by ghostdogpr — APPROVED** [review]

**Review by sideeffffect — APPROVED** [review]
Nice, thanks

**Comment by sideeffffect** [comment]
Published in https://github.com/zio/zio-prelude/releases/tag/v1.0.0-RC24

**Pull request merged**